[DOCS-5421] Remote Configuration docs for GA (#18094)
DataDog/documentation: content/en/agent/guide/how_remote_config_works.md

## Security Considerations

Datadog implements the following safeguards to protect the confidentiality, integrity, and availability of configurations received and applied by your Datadog components:

* Agents deployed in your infrastructure request configurations from Datadog.
* Datadog never sends configurations unless requested by Agents, and only sends configurations relevant to the requesting Agent.
* Because the configuration requests are initiated from your Agents to Datadog over HTTPS (port 443), there is no need to open additional ports in your network firewall.
* The communication between your Agents and Datadog is encrypted using HTTPS, and is authenticated and authorized using your Datadog API key.

> * Only users with the [`api_keys_write`][3] permissions are authorized to enable or disable Remote Configuration capability on the API key and use the supported product features.

Reviewer: Above this line, can we also add below?

Only users with the [org_management](https://docs-staging.datadoghq.com/aliciascott/how_rc_works/account_management/rbac/permissions/#access-management)  permission are authorized to enable or disable Remote Configuration at the organization level.

Reviewer 2: Is this something you want to highlight under a separate "Permissions" section?

Author: This was actually already there on the latest staging doc so I'll mark this as done.

```yaml
datadog:
  apiKey: xxx
  remoteConfiguration:
    enabled: true
```

{{% /tab %}}
{{< /tabs >}}

6. Restart your Agent for the changes to take effect. 

After you perform these steps, your Agent requests its configuration from Datadog, and the features that use remote configuration are enabled:
- [CWS default agent rules][5] update automatically as released.
- [Datadog Remote Instrumentation][13] is enabled.
- [APM Agent-level sampling rates][6] are applied.  
- [Dynamic Instrumentation][7] is enabled.
- [ASM 1-Click enablement, IP blocking, and attack pattern updates][8] are enabled.

## Best practices

### Datadog Audit Trail

Use [Datadog Audit Trail][14] to monitor organization access and Remote Configuration enabled events. Audit Trail allows your administrators and security teams to track the creation, deletion, and modification of Datadog API and application keys. After Audit Trail is configured, you can view events related to Remote Configuration enabled features and who has requested these changes. Audit Trail allows you to reconstruct sequences of events, and establish robust Datadog monitoring for Remote Configuration. 

> ### Monitors

Reviewer 2: ```suggestion

### Monitors
```

DD_REMOTE_CONFIGURATION_ENABLED=true
```

{{% /tab %}}
{{% tab "Helm" %}}
Add the following to your Helm chart, specifying the API key that has Remote Config capability enabled:
```yaml
datadog:

>   apiKey:  # <DATADOG_API_KEY>

Author: Noted! I'll fix that.

- [Datadog Remote Instrumentation][13] is enabled.
- [APM Agent-level sampling rates][6] are applied.  
- [Dynamic Instrumentation][7] is enabled.
- [ASM 1-Click enablement, IP blocking, and attack pattern updates][8] are enabled.

## Best practices

### Datadog Audit Trail

Use [Datadog Audit Trail][14] to monitor organization access and Remote Configuration enabled events. Audit Trail allows your administrators and security teams to track the creation, deletion, and modification of Datadog API and application keys. After Audit Trail is configured, you can view events related to Remote Configuration enabled features and who has requested these changes. Audit Trail allows you to reconstruct sequences of events, and establish robust Datadog monitoring for Remote Configuration. 
### Monitors
Configure [monitors][15] to receive notifications when an event of interest is encountered.

## Troubleshooting

### Restart the Agent

After the Agent configuration is updated in the [`datadog.yaml`][16] file, restart the Agent for this change to take effect. 

### Ensure Datadog Config endpoints are reachable from your environment	

> To perform Remote Configuration, both the Agent and  the Observability Pipelines Worker deployed in your environment communicate to Datadog Config [endpoints][17]. Ensure that outbound HTTPS has access to these endpoints from your environment. If you also have a proxy in between Datadog and your environment, update your proxy settings to incorporate Config destinations.

Author: Hey @estherk15 there is an extra space here and I am not sure how to get past this error, if I pull the changes locally, its out of date and won't let me update it because it shows my branch is behind. How can I just remove the extra space and get this check to pass ?

> To perform Remote Configuration, the Agent and Observability Pipelines Worker deployed in your environment communicate to Datadog Config [endpoints][17]. Ensure you are allowing outbound HTTPS access to these endpoints from your environment. If you also have a proxy in between Datadog and your environment, update your proxy settings to incorporate Config destinations.

Author: Good point, I'll add a link to Remote Config endpoint destinations there like I did above and change the wording slightly.

### Enable Remote Configuration at the Organization level

> To enable Remote Configuration at the [Organization][4] level in the Datadog UI, follow the **Organization Settings > Security > Remote Configuration** menu. This allows your authenticated and authorized Datadog components to remotely receive configurations and security detection rules of supported features from Datadog. Only users who have the [`org_management`][9] RBAC permission are able to enable Remote Configuration at the Organization level.

Reviewer 2: ```suggestion
To enable Remote Configuration at the [Organization][4] level in the Datadog UI, follow the **Organization Settings > Security > Remote Configuration** menu. This allows your authenticated and authorized Datadog components to remotely receive configurations and security detection rules of supported features from Datadog. Only users who have the [`org_management`][9] RBAC permission can enable Remote Configuration at the Organization level.
```

### Enable Remote Configuration on the API key

> To authenticate and authorize the Agent to receive configurations and security detection rules, and to allow the Observability Pipelines Worker to receive configurations, enable Remote Configuration on the relevant API Key. Only users who have the [`api_keys_write`][3] RBAC permission are able to enable Remote Configuration on the API Key.

Reviewer 2: ```suggestion
To authenticate and authorize the Agent to receive configurations and security detection rules, and to allow the Observability Pipelines Worker to receive configurations, enable Remote Configuration on the relevant API Key. Only users who have the [`api_keys_write`][3] RBAC permission can enable Remote Configuration on the API Key.
```

> **Note:** If you have [`api_keys_write`][3] RBAC permission, but are missing Remote Configuration [Organization][4] level permissions, the ability to enable Remote Configuration on a new or an existing API Key is not available. You would only have permission to disable Remote Configuration on an existing API Key.

Reviewer 2: ```suggestion
**Note:** If you have [`api_keys_write`][3] RBAC permission, but are missing Remote Configuration [Organization][4] level permissions, you cannot enable Remote Configuration on a new or an existing API Key. You only have permission to disable Remote Configuration on an existing API Key.
```

### Review Remote Configuration status events

> To provide visibility into the Remote Configuration status of your Agent, Datadog has added Remote Configuration status' in the [Remote Configuration UI][4]. The following chart describes the meaning of each status:

Reviewer 2: ```suggestion
Gain visibility into the Remote Configuration status of your Agent through the Remote Configuration statuses in the [Remote Configuration UI][4]. The following table describes the meaning of each status:
```
